### PR TITLE
Screen Recorders integration.

### PR DESCRIPTION
**Pull request by MateusX02**

### CODE REVIEW

**Comment by tposejank** [comment]
no

**Comment by MateusX02** [comment]
ok

**Comment by MateusX02** [comment]
¯\_(ツ)_/¯

File: source/MainMenuState.hx [hunk 1]
```diff
@@ -199,7 +209,7 @@ class MainMenuState extends MusicBeatState
 			{
 				if (optionShit[curSelected] == 'donate')
 				{
-					CoolUtil.browserLoad('https://ninja-muffin24.itch.io/funkin');
+					.CoolUtilbrowserLoad('https://ninja-muffin24.itch.io/funkin');
```
Comment (Steviegt6) [inline source/MainMenuState.hx:212]: This won't compile, you should test changes before committing.

File: source/MainMenuState.hx [hunk 2]
```diff
@@ -171,6 +175,12 @@ class MainMenuState extends MusicBeatState
 			FlxG.sound.music.volume += 0.5 * FlxG.elapsed;
 		}
 
+		if (CoolUtil.isRecording())
+		{
+			trace('he's recording da game bro"');
```
Comment (Steviegt6) [inline source/MainMenuState.hx:180]: Invalid string declaration, should be self-explanatory.

File: source/MainMenuState.hx [hunk 3]
```diff
@@ -125,11 +128,11 @@ class MainMenuState extends MusicBeatState
 
 		FlxG.camera.follow(camFollowPos, null, 1);
 
-		var versionShit:FlxText = new FlxText(12, ClientPrefs.getResolution()[1] - 44, 0, "Psych Engine v" + psychEngineVersion, 12);
+		var versionShit:FlxText = new FlxText(12, FlxG.height - 44, 0, "Psych Engine v" + psychEngineVersion, 12);
 		versionShit.scrollFactor.set();
 		versionShit.setFormat("VCR OSD Mono", 16, FlxColor.WHITE, LEFT, FlxTextBorderStyle.OUTLINE, FlxColor.BLACK);
 		add(versionShit);
-		var versionShit:FlxText = new FlxText(12, ClientPrefs.getResolution()[1] - 24, 0, "Friday Night Funkin' v" + Application.current.meta.get('version'), 12);
+		var versionShit:FlxText = new FlxText(12, FlxG.height - 24, 0, "Friday Night Funkin' v" + Application.current.meta.get('version'), 12);
```
Comment (Steviegt6) [inline source/MainMenuState.hx:135]: Redundant change?

File: source/MainMenuState.hx [hunk 4]
```diff
@@ -139,6 +142,7 @@ class MainMenuState extends MusicBeatState
 		changeItem();
 
 		#if ACHIEVEMENTS_ALLOWED
+		Achievements.loadAchievements();
```
Comment (Steviegt6) [inline source/MainMenuState.hx:145]: Redundant change?

File: source/MainMenuState.hx [hunk 5]
```diff
@@ -68,10 +70,12 @@ class MainMenuState extends MusicBeatState
 
 		persistentUpdate = persistentDraw = true;
 
+		FlxG.mouse.visible = true;
+
 		var yScroll:Float = Math.max(0.25 - (0.05 * (optionShit.length - 4)), 0.1);
 		var bg:FlxSprite = new FlxSprite(-80).loadGraphic(Paths.image('menuBG'));
 		bg.scrollFactor.set(0, yScroll);
-		bg.setGraphicSize(Std.int(bg.width * 1.175*scaleRatio ));
+		bg.setGraphicSize(Std.int(bg.width * 1.175));
```
Comment (Steviegt6) [inline source/MainMenuState.hx:78]: Redundant change? The mouse doesn't need to be visible.

I know that `scaleRatio` needs an "overhaul" due to compilation targets, but not here.

File: source/MainMenuState.hx [hunk 6]
```diff
@@ -84,14 +88,13 @@ class MainMenuState extends MusicBeatState
 
 		magenta = new FlxSprite(-80).loadGraphic(Paths.image('menuDesat'));
 		magenta.scrollFactor.set(0, yScroll);
-		magenta.setGraphicSize(Std.int(magenta.width * 1.175*scaleRatio ));
+		magenta.setGraphicSize(Std.int(magenta.width * 1.175));
```
Comment (Steviegt6) [inline source/MainMenuState.hx:91]: Ditto.

File: source/MainMenuState.hx [hunk 7]
```diff
@@ -16,6 +16,7 @@ import flixel.math.FlxMath;
 import flixel.tweens.FlxEase;
 import flixel.tweens.FlxTween;
 import flixel.util.FlxColor;
+import flixel.ui.FlxButton;
```
Comment (Steviegt6) [inline source/MainMenuState.hx:19]: Redundant import.

File: source/CoolUtil.hx [hunk 8]
```diff
@@ -116,9 +150,9 @@ class CoolUtil
 
 	//uhhhh does this even work at all? i'm starting to doubt
 	public static function precacheSound(sound:String, ?library:String = null):Void {
-		var EmbeddedSound = Paths.sound(sound, library);
-		if (Assets.exists(EmbeddedSound, SOUND) || Assets.exists(EmbeddedSound, MUSIC))
-			Assets.getSound(EmbeddedSound, true);
+		if(!Assets.cache.hasSound(Paths.sound(sound, library))) {
+			FlxG.sound.cache(Paths.sound(sound, library));
+		}
```
Comment (Steviegt6) [inline source/CoolUtil.hx:155]: Irrelevant to this PR.

File: source/CoolUtil.hx [hunk 9]
```diff
@@ -16,11 +16,45 @@ using StringTools;
...
+
 	public static var defaultDifficulties:Array<String> = [
 		'Easy',
 		'Normal',
 		'Hard'
 	];
+
+	public static function isRecording():Bool
+	{
+		#if FEATURE_OBS
```
Comment (Steviegt6) [inline source/CoolUtil.hx:38]: This should be pre-defined in `project.xml`, typically.

**Review by steviegt6 — CHANGES_REQUESTED** [review]
This entire feature is useless, but this code doesn't even compile.

**Comment by Stilic** [comment]
nononononononononoono NO

**Comment by ActualMandM** [comment]
thanks for taking from DDTO
https://github.com/Jorge-SunSpirit/Doki-Doki-Takeover/blob/41bd9ebd683789d742042e116709245a53e177fa/source/CoolUtil.hx#L39

File: source/MainMenuState.hx (hunk 2 above)
Comment (ActualMandM) [reply source/MainMenuState.hx:180]: this piece of code is also gonna make a command prompt window pop up *every* time the user goes to the main menu state

**Comment by ActualMandM** [comment]
should mention that specifically this version of the OBS checking code was taken from DDTO, hence hypercam 2 being a part of the program list and the FEATURE_OBS define

https://github.com/Jorge-SunSpirit/Doki-Doki-Takeover/blob/41bd9ebd683789d742042e116709245a53e177fa/source/CoolUtil.hx#L39

**Comment by HiroMizuki** [comment]
i think we don't need it

**Pull request closed without merging**